packages/region/src/services/region-module.ts

```diff
@@ -104,6 +104,42 @@ export default class RegionModuleService<
     )
   }
 
+  private validateAndSetCountriesForRegion(
+    region: CreateRegionDTO,
+    countriesInDb: Map<string, TCountry>,
+    countriesToAdd: string[]
+  ) {
+    const notInDb = countriesToAdd.filter(
+      (c) => !countriesInDb.has(c.toLowerCase())
+    )
+
+    if (notInDb.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Countries with codes ${countriesToAdd.join(", ")} does not exist`
+      )
+    }
+
+    const regionCountries = countriesToAdd.map((code) => {
+      const country = countriesInDb.get(code.toLowerCase())
+
+      if (country?.region_id) {
```

Comment:
This should always exist at this point given the validation above right?
```suggestion
      if (country!.region_id) {
```

Reply:
Which validation point?

First is checking if the country exists e.g. if you try to add Wonderland it will throw. 

Second is checking if you are assigning a country, that's already assigned to a region e.g. if you create a region "US" with country "us", but region "North America" already exists (and includes "us").

Reply:
@pKorsholm, ping :D

Reply:
I meant you know the country is in the database at this point, you've validated that the country exists using the `notInDb` check, i.e. the map will never return undefined. It's a nit comment at best, feel free to disregard 😄